api/src/main/java/com/github/retrooper/packetevents/util/PEVersion.java

```diff
@@ -18,142 +18,201 @@
 
 package com.github.retrooper.packetevents.util;
 
-import java.util.Arrays;
+import com.github.retrooper.packetevents.PacketEvents;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.Optional;
 
 /**
- * PacketEvents version.
- * This class represents a PacketEvents version.
- *
- * @author retrooper
- * @since 1.8
+ * Represents a PacketEvents version using Semantic Versioning.
+ * Supports version comparison, cloning, and provides a string representation.
  */
-public class PEVersion {
+public class PEVersion implements Comparable<PEVersion> {
+
+    private final int major;
+    private final int minor;
+    private final int patch;
+    private final boolean snapshot;
+
     /**
-     * Array containing the digits in the version.
-     * For example, "1.8.9" will be stored as {1, 8, 9} in an array.
+     * Constructs a {@link PEVersion} instance.
+     *
+     * @param major    the major version number.
+     * @param minor    the minor version number.
+     * @param patch    the patch version number.
+     * @param snapshot whether the version is a snapshot.
      */
-    private final int[] versionIntArray;
+    public PEVersion(final int major, final int minor, final int patch, final boolean snapshot) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.snapshot = snapshot;
+    }
 
     /**
-     * Specify your version using an array.
+     * Constructs a {@link PEVersion} instance with snapshot defaulted to false.
      *
-     * @param version Array version.
+     * @param major the major version number.
+     * @param minor the minor version number.
+     * @param patch the patch version number.
      */
-    public PEVersion(final int... version) {
```

Comment:
this constructor should be re-added, otherwise this would be a breaking change

Reply:
> this constructor should be re-added, otherwise this would be a breaking change

Shall I mark it as deprecated, and then make it use the overload constructor which defaults snapshot to false?

Reply:
Oops, missed your response - yeah, imo thats a good solution